drivers/adc/adc_stm32.c

```diff
@@ -811,6 +815,13 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VREFINT) == channel_id) {
 		adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_VREFINT);
 	}
+#if defined(CONFIG_STM32_VBAT)
+	/* Enable the bridge divider only when needed for ADC conversion. */
+	if (config->has_vbat_channel &&
+		__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_VBAT) == channel_id) {
+		adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_VBAT);
```

Comment:
This line will set the common path for `LL_ADC_PATH_INTERNAL_VBAT`, which connects the VBAT pin to the internal voltage divider. According to the reference manual, the connection should only be made during the sampling, and VBAT should be disconnected from the voltage divider after sampling to avoid power drain.

I think something should call `LL_ADC_SetCommonPathInternalChRem()` at the end of `start_read` or in `adc_stm32_read` after `start_read` to remove `LL_ADC_PATH_INTERNAL_VBAT`